src/Libraries/Microsoft.Extensions.AI.Abstractions/ChatCompletion/StreamingChatCompletionUpdateExtensions.cs

```diff
@@ -136,17 +136,33 @@ private static void AddMessagesToCompletion(Dictionary<int, ChatMessage> message
     {
         if (messages.Count <= 1)
         {
+            // Add the single message if there is one.
             foreach (var entry in messages)
             {
                 AddMessage(completion, coalesceContent, entry);
             }
+
+            // In the vast majority case where there's only one choice, promote any additional properties
+            // from the single message to the chat completion, making them more discoverable and more similar
+            // to how they're typically surfaced from non-streaming services.
+            if (completion.Choices.Count == 1 &&
+                completion.Choices[0].AdditionalProperties is { } messageProps)
+            {
+                completion.Choices[0].AdditionalProperties = null;
+                completion.AdditionalProperties = messageProps;
+            }
```

Comment:
If the result sometimes has one choice and sometimes has multiple choices, it may become difficult to know where to find a particular "AdditionalProperties" value. Someone who needs to access it programmatically would have to replicate this logic (i.e., if there's one choice look in one place, otherwise look in another place).

Maybe that's acceptable or can never happen for some reason.

Is there a high payoff from moving the `AdditionalProperties` over?

Reply:
> Is there a high payoff from moving the AdditionalProperties over?

In the majority case having the same behavior with regards to additional properties between a ChatCompletion from CompleteAsync and one from ToChatCompletion (from a list of streaming updates). Otherwise, with the former top-level properties show up on the ChatCompletion but with the latter they show up on the ChatMessage.

Reply:
> If the result sometimes has one choice and sometimes has multiple choices

That also seems very unlikely for a given call site. It's possible someone could variable set n, but that would be an explicit action that would already impact things based on one vs multiple choices, like whether automatic function invocation works.